src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    static final int MAX_SIZE = 10;
+    static final double GROW_COEFFICIENT = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (array.length == size) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index do not exist");
+        }
+        if (array.length == size) {
+            grow();
+        }
+        fastValueAdd(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            array[size] = list.get(i);
+            size++;
```

Comment:
you should use the method `add(T value)` here, cause if you try to add more elements than you have in the array.length, you will get ArrayIndexOutOfBoundsException
```suggestion
            add(list.get(i));
```